From d9c1eed4039e66a17cd64d726b2daf448b0699c7 Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Mon, 2 Sep 2013 16:29:13 +0800
Subject: [PATCH 0154/1074] ENGR00275034-4 ARM: dts: add camera ov5640 support
 on imx6sl-evk

Add ov5640 camera support on imx6sl-evk
Add binding document for csi/csi-v4l2-capture/ov5640

Signed-off-by: Robby Cai <R63905@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 .../bindings/video/fsl,csi-v4l2-capture.txt        |   61 ++++++++++++++++++++
 arch/arm/boot/dts/imx6sl-evk.dts                   |   16 +++++
 2 files changed, 77 insertions(+), 0 deletions(-)
 create mode 100644 Documentation/devicetree/bindings/video/fsl,csi-v4l2-capture.txt

diff --git a/Documentation/devicetree/bindings/video/fsl,csi-v4l2-capture.txt b/Documentation/devicetree/bindings/video/fsl,csi-v4l2-capture.txt
new file mode 100644
index 0000000..abfe6f4
--- /dev/null
+++ b/Documentation/devicetree/bindings/video/fsl,csi-v4l2-capture.txt
@@ -0,0 +1,61 @@
+* Freescale CMOS Sensor Interface (CSI) V4L2 Capture
+
+Required properties for CSI
+- compatible: "fsl,<soc>-csi". Supported chip includes imx6sl
+- reg: Address and length of the register set for CSI
+- interrupts: Should contain CSI interrupts
+
+Required properties for v4l2_capture
+- compatible: should be "fsl,<soc>-csi-v4l2", supported socs include imx6sl
+
+Required properties for sensor
+- compatible:  "<vendor>,<sensor>"
+  please check the supported sensor in the Supported Sensor fields.
+- reg: sensor I2C slave address
+- pinctrl-names: should be "default" for parallel sensor
+- pinctrl-0: should depend on the connection between sensor and i.MX
+  connection between sensor and i.MX could be only legacy parallel on i.MX6SL
+- clocks: should be the clock source provided to sensor.
+- clock-names: should be "csi_mclk"
+- AVDD-supply: set according to the board.
+- DVDD-supply: set according to the board.
+- pwn-gpios: set according to the board.
+- rst-gpios: set according to the board.
+- csi_id: csi id for v4l2 capture device
+  should be 0 for i.MX6SL
+- mclk: should the value of mclk clock send out the sensor. unit is Hz.
+- mclk_source: should be 0 for i.MX6SL
+
+Supported Sensor
+- ovti, ov5640
+
+Example for CSI:
+	csi: csi@020e4000 {
+		compatible = "fsl,imx6sl-csi";
+		reg = <0x020e4000 0x4000>;
+		interrupts = <0 7 0x04>;
+		status = "disabled";
+	};
+
+Examples for v4l2_capture:
+	csi_v4l2_cap {
+		compatible = "fsl,imx6q-v4l2-capture";
+		status = "okay";
+	};
+
+Examples for sensors:
+	ov564x: ov564x@3c {
+		compatible = "ovti,ov564x";
+		reg = <0x3c>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_csi_0>;
+		clocks = <&clks IMX6SL_CLK_CSI>;
+		clock-names = "csi_mclk";
+		AVDD-supply = <&vgen6_reg>;  /* 2.8v */
+		DVDD-supply = <&vgen2_reg>;  /* 1.5v*/
+		pwn-gpios = <&gpio1 25 1>;
+		rst-gpios = <&gpio1 26 0>;
+		csi_id = <0>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+	};
diff --git a/arch/arm/boot/dts/imx6sl-evk.dts b/arch/arm/boot/dts/imx6sl-evk.dts
index 2830479..b0f6a24 100644
--- a/arch/arm/boot/dts/imx6sl-evk.dts
+++ b/arch/arm/boot/dts/imx6sl-evk.dts
@@ -421,6 +421,22 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c3_1>;
 	status = "okay";
+
+	ov564x: ov564x@3c {
+		compatible = "ovti,ov564x";
+		reg = <0x3c>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_csi_0>;
+		clocks = <&clks IMX6SL_CLK_CSI>;
+		clock-names = "csi_mclk";
+		AVDD-supply = <&vgen6_reg>;  /* 2.8v */
+		DVDD-supply = <&vgen2_reg>;  /* 1.5v*/
+		pwn-gpios = <&gpio1 25 1>;
+		rst-gpios = <&gpio1 26 0>;
+		csi_id = <0>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+	};
 };
 
 &iomuxc {
-- 
1.7.5.4

